pkg/metrics/controller.go

```diff
@@ -262,9 +269,43 @@ func (c *Controller) recordGameServerStatusChanges(old, next interface{}) {
 		}
 		recordWithTags(context.Background(), []tag.Mutator{tag.Upsert(keyType, string(newGs.Status.State)),
 			tag.Upsert(keyFleetName, fleetName)}, gameServerTotalStats.M(1))
+
+		// Calculate the duration from the start of the GameServer
+		c.stateLock.Lock()
+		defer c.stateLock.Unlock()
+		err := calcDuration(newGs, oldGs)
+		if err != nil {
+			c.logger.Info(err.Error())
+		}
 	}
 }
 
+func calcDuration(newGs, oldGs *agonesv1.GameServer) error {
+	fleetName := newGs.Labels[agonesv1.FleetNameLabel]
+	diff := time.Now().Sub(newGs.ObjectMeta.CreationTimestamp.Local()).Seconds()
+	key := newGs.ObjectMeta.Name + "/" + string(newGs.Status.State)
+	oldKey := oldGs.ObjectMeta.Name + "/" + string(oldGs.Status.State)
+	if !GameServerStateLastChange.Contains(key) {
+		GameServerStateLastChange.Add(key, diff)
+		recordWithTags(context.Background(), []tag.Mutator{tag.Upsert(keyType, string(newGs.Status.State)),
+			tag.Upsert(keyFleetName, fleetName)}, gsStateDuration.M(diff*1000.))
```

Comment:
Should it be "*1000"? Check what .Seconds() returns, seems no need to multiply diff here